sympy/solvers/ode/ode.py

```diff
@@ -1752,6 +1752,209 @@ def match_2nd_linear_bessel(r, func):
+                    if k == 0:
+                        coef = eq.as_independent(func, as_Add=True)[1]
+                        for xr in range(1, ode_order(eq, func) + 1):
+                            coef -= eq.as_independent(diff(func, t, xr), as_Add=True)[1]
+                        if coef != 0:
+                            is_linear_ = False
+                    else:
+                        if eq.as_independent(diff(func, t, k), as_Add=True)[1]:
+                            is_linear_ = False
+                else:
+                    for func_ in funcs:
+                        if isinstance(func_, list):
+                            for elem_func_ in func_:
+                                dep = coeffs[diff(func, t, k)].as_independent(elem_func_, as_Add=True)[1]
+                                if dep != 0:
+                                    is_linear_ = False
+                        else:
+                            dep = coeffs[diff(func, t, k)].as_independent(func_, as_Add=True)[1]
+                            if dep != 0:
+                                is_linear_ = False
+    if is_linear_:
+        return coeffs
+    return None
+
+
+def neq_nth_linear_constant_coeff_match(eqs, funcs):
+    from sympy.solvers.solveset import linear_eq_to_matrix
+
+    match = dict()
+    match["no_of_equation"] = len(eqs)
+    match["eq"] = eqs
+
+    # Error for i == 0 can be added but isn't for now
+
+    # Getting the independent variable
+    t = list(list(eqs[0].atoms(Derivative))[0].atoms(Symbol))[0]
+
+    # Assuming funcs has all the funcs mentioned and
+    # len(funcs) == len(eqs)
+
+    # Getting the func_dict and order using the helper
+    # function
+    funcs, order = _get_func_order(eqs, funcs)
+    match['func'] = funcs
+    match['order'] = order
+
+    # Not adding the check if the len(func.args) for
+    # every func in funcs is 1
+
+    # Getting coefficient matrix and rhs matrix(A and
+    # b)
+    terms = []
+    rep = {}
+    for func in funcs:
+        temp_terms = []
+        for k in range(order[func] + 1):
+            globals()["{}{}".format(type(func), k)] = symbols("{}{}".format(type(func), k))
+            rep[diff(func, t, k)] = globals()["{}{}".format(type(func), k)]
+            temp_terms += [globals()["{}{}".format(type(func), k)]]
+        terms += temp_terms
+    eqs_sub = [eq.subs(rep) for eq in eqs]
+
+    # Linearity check
+    try:
+        A, b = linear_eq_to_matrix(eqs_sub, terms)
+        is_linear = True
+
+        # Constant coefficient check
+        is_constant = True
+        for coef in A:
+            if is_constant == True:
+                if coef.as_independent(t, as_Add=True)[1] != 0:
+                    is_constant = False
+
+        # Homogeneous check
+        is_homogeneous = True
+        for rhs in b:
+            if is_homogeneous == True:
+                if rhs != 0:
+                    is_homogeneous = False
+
+    except ValueError:
```

Comment:
Exceptions should be caught around the narrowest block of code possible. If this is intended to catch the error from `linear_eq_to_matrix` then only that call should be in the try block.